keras/preprocessing/image.py

```diff
@@ -437,6 +437,50 @@ def fit(self, X,
             U, S, V = linalg.svd(sigma)
             self.principal_components = np.dot(np.dot(U, np.diag(1. / np.sqrt(S + 10e-7))), U.T)
 
+    def transform(self,x, train_mode):
+        '''
+        Transforms single image.
+        Required for featurewise_center, featurewise_std_normalization
+        and zca_whitening.
+
+        # Arguments
+            X: Numpy array, the data to fit on.
+            augment: whether to fit on randomly augmented samples
+            rounds: if `augment`,
+                how many augmentation passes to do over the data
+            seed: random seed.
+
+        Arguments:
+            x: image(Numpy array) to transform
+            train_mode: True if random augmentation should be applied, False - steps needed to be applied on tst and train data(aka normalisation)
+
+        Returns: transformed image
+        '''
+        if train_mode:
+            x = self.random_transform(x)
+        return self.standardize(x)
```

Comment:
I am not that experienced in image processing(thanks to this library I understand some cool DL concepts and image ideas). I could miss something but after reading some papers  about building models such as VGG16/18 I got idea that you do random transformation to artificially increase number of training examples and therefore prevent overfitting and have ability to use more complex models.  So they usually do random transformation for training examples and apply normalization/standardization to eval and test data. I guess you right that using random transformation on eval and testing is common, but I also think it is common not to do them on test data. And data preprocessing is always have to be applied to train and test, because you don't want train on mean subtracted data and apply it to test. 

I am thinking about something like transformers in sklearn, but with DL specific you might want to have slightly different transformation on test/train. And with two options train_mode=True/False it cover all use cases.